### PR TITLE
Big change to starship locks, will now stay locked 10minutes after ex…

### PR DESCRIPTION
**Pull request by SciathGH**
…peration and unlocks after. No option to disable it in menu.
I cant test in my test server, so warning in advance.

### CODE REVIEW

**Comment by trainb0y1** [comment]
Out of curiosity what’s the reasoning behind this? Isn’t the point of ship locks to prevent ship theft while the owner is offline?

**Comment by SciathGH** [comment]
No its for combat releasing, per Peters request

**Comment by trainb0y1** [comment]
Ok. Just curious lol

File: src/main/kotlin/net/starlegacy/feature/starship/PilotedStarships.kt [hunk 1]
```diff
@@ -254,7 +256,6 @@ object PilotedStarships : SLComponent() {
 		player.sendFeedbackActionMessage(SUCCESS, "Released {0}", getDisplayName(starship.data))
 		return true
 	}
-
```
Comment (Peter-Crawley) [inline src/main/kotlin/net/starlegacy/feature/starship/PilotedStarships.kt:257]: This newline should stay.

File: src/main/kotlin/net/starlegacy/feature/starship/StarshipComputers.kt [hunk 2]
```diff
@@ -86,7 +86,6 @@ object StarshipComputers : SLComponent() {
 		if (data == null) {
 			return
 		}
-
```
Comment (Peter-Crawley) [inline src/main/kotlin/net/starlegacy/feature/starship/StarshipComputers.kt:89]: This newline should stay.

File: src/main/kotlin/net/starlegacy/feature/starship/PilotedStarships.kt [hunk 3]
```diff
@@ -146,6 +146,8 @@ object PilotedStarships : SLComponent() {
 		val pilotedStarship = PilotedStarships[player]
 		if (pilotedStarship != null) {
 			if (pilotedStarship.dataId == data._id) {
+				DeactivatedPlayerStarships.updateLockEnabled(data, true)
+				player.sendFeedbackMessage(INFORMATION, "Your starship will unlock after {0}minutes", PlayerStarshipData.LOCK_TIME_MS/60000)
```
Comment (Peter-Crawley) [inline src/main/kotlin/net/starlegacy/feature/starship/PilotedStarships.kt:150]: `{0} minutes.`

File: src/main/kotlin/net/starlegacy/database/schema/starships/PlayerStarshipData.kt [hunk 4]
```diff
@@ -73,6 +73,8 @@ data class PlayerStarshipData(
 
 	/** assumes that it's also deactivated */
 	fun isLockActive(): Boolean {
-		return isLockEnabled && System.currentTimeMillis() - lastUsed >= LOCK_TIME_MS
+		return if (System.currentTimeMillis()-lastUsed>LOCK_TIME_MS) {
```
Comment (Peter-Crawley) [inline src/main/kotlin/net/starlegacy/database/schema/starships/PlayerStarshipData.kt:76]: This checks if it has been more then `LOCK_TIME_MS` not less then it.

File: src/main/kotlin/net/starlegacy/database/schema/starships/PlayerStarshipData.kt [hunk 5]
```diff
@@ -48,7 +48,7 @@ data class PlayerStarshipData(
 		ensureIndex(PlayerStarshipData::world)
 		ensureUniqueIndex(PlayerStarshipData::world, PlayerStarshipData::blockKey)
 	}) {
-		const val LOCK_TIME_MS = 1_000 * 5
+		const val LOCK_TIME_MS = 60_000 * 10
```
Comment (Peter-Crawley) [inline src/main/kotlin/net/starlegacy/database/schema/starships/PlayerStarshipData.kt:51]: If this value is only used once then remove it and just hard code it.

**Review by Peter-Crawley — CHANGES_REQUESTED** [review]

File: src/main/kotlin/net/starlegacy/database/schema/starships/PlayerStarshipData.kt (hunk 5 above)
Comment (SciathGH) [reply src/main/kotlin/net/starlegacy/database/schema/starships/PlayerStarshipData.kt:51]: Its used twice

Comment (Peter-Crawley) [reply src/main/kotlin/net/starlegacy/database/schema/starships/PlayerStarshipData.kt:51]: Okay fair enough.

**Pull request closed without merging**